### PR TITLE
Initial contributing guide

### PR DESCRIPTION
**Pull request by MathiasMagnus**

### CODE REVIEW

**Review by saadrahim — CHANGES_REQUESTED** [review]
Add to the about section via _toc.yml.in

File: docs/.sphinx/requirements.in
Comment (MathiasMagnus) [reply docs/.sphinx/requirements.in:1]: Restored, as per our conversation.

File: CONTRIBUTING.md
Comment (MathiasMagnus) [reply CONTRIBUTING.md]: Done.

Comment (MathiasMagnus) [reply CONTRIBUTING.md]: Did.

File: CONTRIBUTING.md
Comment (MathiasMagnus) [reply CONTRIBUTING.md]: Explained, port override removed.

File: CONTRIBUTING.md
Comment (AlexVlx) [inline CONTRIBUTING.md:89]: How would this look if we made it a `>` quote block, rather than italicised? Also, we've tried to be relatively disciplined elsewhere in using `*` to denote styling, rather than `_`, so it might be nice to be uniform.

File: CONTRIBUTING.md
Comment (AlexVlx) [inline CONTRIBUTING.md:36]: Indentation is a bit wonky here, I think, we should do either 2 or 4 spaces. Perhaps 2 on this occasion because this is pretty spammy and overflows when rendered? Your call though.

**Review by AlexVlx — COMMENTED** [review]
I think it would be desirable to keep line length under 80 characters, except when there's absolutely no way to stick to that. This ensures side-by-side viewing is always available and efficient (diffs, redering etc.)

File: CONTRIBUTING.md
Comment (MathiasMagnus) [reply CONTRIBUTING.md:36]: Well, the first 4 spaces is for Markdown to indent the content to the same level as the numbered list (for aesthetics). Then, within the content I used tab to designate indentation and leave it up the the renderer to decide how many characters that is. It's true, that it could be uniform, Markdown would understand the top level spaces be tabs as well.

The thing is, that for Markdown to indent the content, it needs 3-4 spaces in the outermost level (or a tab) to indent. Would one tab then 2 spaces per-level work?

File: CONTRIBUTING.md
Comment (saadrahim) [inline CONTRIBUTING.md:5]: We need community guidelines. I will work to set community guidelines. This section can stay until we adopt community guidelines.

**Comment by saadrahim** [comment]
> Add to the about section via _toc.yml.in

This needs to be addressed. Please work to get this in the HTML build of the documentation.

File: CONTRIBUTING.md
Comment (AlexVlx) [reply CONTRIBUTING.md]: Indenting with the list is fine, however, it's the indenting within the codeblock itself that lloked a bit off, see e.g. the opening curly for the `tasks` array.

**Comment by saadrahim** [comment]
Added some feedback at https://github.com/StreamHPC/ROCm/pull/1

**Comment by saadrahim** [comment]
> Can you add a section on our use of GitHub markdown? https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/quickstart-for-writing-on-github
> 
> We can also add a rocm specific section that ROCm documentation only contains one level one header in md and rst files.
> 
> There is a lot of toolchain discussion in the about page. It should be restructured to fit the changes that you are making here.
> 
> Also, be prepared to amend the document for #1946 and #1947
> 
> The standard encourage contributors statement is needed. Add statement to request help via the github discussion if there are issues with contributing.
> 
> @AlexVlx Anything else to go in the contributors guide?

I've addressed most of these statements myself in my PR to your branch. The PR supersedes these comments.

File: CONTRIBUTING.md
Comment (MathiasMagnus) [reply CONTRIBUTING.md]: Should look better now.

**Comment by MathiasMagnus** [comment]
@saadrahim Should be in the TOC now too.

**Review by saadrahim — APPROVED** [review]

**Comment by AlexVlx** [comment]
@MathiasMagnus overall this LGTM, modulo one aspect which is line length - could you please keep line length at 80 characters maximum? This might sound excessively old school, but for things like .md it is really useful to be able to roll the source and the rendered preview side by side with no wraparaound. Thanks!

**Comment by MathiasMagnus** [comment]
@AlexVlx Line length addressed. Will be enforced via #1946

**Review by AlexVlx — APPROVED** [review]
LGTM, thanks!

**Review by saadrahim — APPROVED** [review]
Making further changes to contribution.md via a future PR. Intend to state MyST is for legacy documentation. Future documentation should be GitHub markdown with exceptions for sphinx-design, breathe (for doxygen), exhale (for doxygen).

**Pull request merged**
